documents/sd/chapter/Classes/Common/Events/Events.tex

```diff
@@ -11,101 +11,314 @@ \subsection*{MORR.Shared.Events}
+	
+	\clsdsp{A simple Exception that encapsulates errors occuring in ChannelStrategies.}
+	
+\end{class}
+	
 \begin{interface}{IEventQueueStorageStrategy}
-    \clsdiagram[width=0.4\textwidth]{resources/Classes/Common/Shared/Events/IEventQueueStorageStrategy.png}
+    \clsdiagram[width=0.5\textwidth]{resources/Classes/Common/Shared/Events/Queue/Strategy/IEventQueueStorageStrategy.png}
 
     \clsdcl{public interface IEventQueueStorageStrategy<T> where T : Event}
 
     \clsdsp{Provides the backing store of an EventQueue with a specific storage strategy.}
 
-    \begin{methods}
-        \begin{method}{IAsyncEnumerable<T> GetEventsInternal()}{Asynchronously gets all events as concrete type T from this storage strategy.}
-            \return{IAsyncEnumerable<T>}{A stream of events of type T.}
-        \end{method}
-        \begin{method}{void Enqueue(T @event)}{Asynchronously enqueues a new event of type T with this storage strategy.}
-            \begin{parameters}
-                \para{T @event}{The event to enqueue.}
-            \end{parameters}
-        \end{method}
-    \end{methods}
 \end{interface}
 
-\begin{class}{RingBufferStorageStrategy}
-    \clsdiagram[width=0.4\textwidth]{resources/Classes/Common/Shared/Events/RingBufferStorageStrategy.png}
+\subsubsection*{MORR.Shared.Event.Queue.Strategy.MultiConsumer}
 
-    \clsdcl{public class RingBufferStorageStrategy<T> : IEventQueueStorageStrategy<T> where T : Event}
+\begin{class}{BoundedMultiConsumerChannelStrategy<TEvent>}
+	\clsdiagram[width=0.5\textwidth]{resources/Classes/Common/Shared/Events/Queue/Strategy/MultiConsumer/BoundedMultiConsumerChannelStrategy.png}
+	
+	\clsdcl{public class BoundedMultiConsumerChannelStrategy<TEvent> : 
+	MultiConsumerChannelStrategy<TEvent> where TEvent : Event}
+	
+	\clsdsp{A BoundedMultiConsumerChannelStrategies is a Distributive-FIFO Queue which allows multiple producers and multiple consumers. However every event is propagated to each consumer once. This queue is bounded by a buffer capacity. To be more detailed each channel is bound by the bufferCapacity. Is the capacity reached for a specific channel it will drop the oldest event and write the new event.}
+	
+	\begin{methods}	
+		\begin{method}{protected override Channel<TEvent> CreateOfferingChannel()}
+			\return{Channel<TEvent>}{}
+		\end{method}
+		\begin{method}{protected override Channel<TEvent> CreateRecievingChannel()}
+			\return{Channel<TEvent>}{}
+		\end{method}
+	\end{methods}
+\end{class}
 
-    \clsdsp{Provides event storage using a ring buffer strategy. That means if the buffer is full, events will be overwritten even if they have not been read by all consumers yet.}
+\begin{absclass}{MultiConsumerChannelStrategy<TEvent>} 
+	\clsdiagram[width=0.5\textwidth]{resources/Classes/Common/Shared/Events/Queue/Strategy/MultiConsumer/MultiConsumerChannelStrategy.png}
+	
+	\clsdcl{public abstract class MultiConsumerChannelStrategy<TEvent> : 
```

Comment:
Probably enforce a newline at the end, as currently the text clips out of the document.